### PR TITLE
docs: improve `docs/reference/types`

### PR DESCRIPTION
**Pull request by Natoandro**
Improves `docs/reference/types`

#### Motivation and context

_N/A_

#### Migration notes

_N/A_

### Checklist

- [ ] The change come with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [x] End-user documentation is updated to reflect the change

### CODE REVIEW

File: website/docs/reference/types/injections.mdx [hunk 1]
```diff
@@ -1,15 +1,126 @@
...
+
+**Parameter:** The value to be set, it must be compatible with the target type.
+
 
 ## Parent
 
-:::caution
+**Description:**
+Parent injection gets the value the output of a sibling field in the parent struct.
+It adds a dependency to the sibling field, so make sure to prevent circular dependencies.
```
Comment (destifo) [inline website/docs/reference/types/injections.mdx:26]: the **gets the value the ouput** sounds a bit confusing here, or am I the only one?

File: website/docs/reference/types/injections.mdx [hunk 2]
```diff
@@ -1,15 +1,126 @@
...
+
+**Method:** `.from_parent(type_name)`
```
Comment (destifo) [inline website/docs/reference/types/injections.mdx:28]: should we also add small snippets examples below the `injection` types, to build the intuition from there? or will the  bundled example would be enough?

**Review by destifo — CHANGES_REQUESTED** [review]
Nice authoring overall 🙌

**Review by Yohe-Am — DISMISSED** [review]

File: website/docs/reference/types/injections.mdx (hunk 2 above)
Comment (afmika) [reply website/docs/reference/types/injections.mdx:28]: I think the bundled example is enough, this would convey  the same concept.

**Review by Yohe-Am — DISMISSED** [review]

**Review by zifeo — APPROVED** [review]

**Review by destifo — APPROVED** [review]

**Pull request merged**